src/NumberController.js

```diff
@@ -65,9 +65,26 @@ export default class NumberController extends Controller {
 
 	_initInput() {
 
+		// Detect if the device has touch as his primary pointer input.
+		// This will targets only phones and tablets.
+		const hasTouchAsPrimaryPointerInput = window.matchMedia( '(pointer: coarse)' ).matches;
```

Comment:
say, do you know if this value changes at runtime? e g if they switch from a finger to a stylus, does it now match `pointer: fine`?

Reply:
No from what I understand the `pointer` media query only returns the _primary_ input.

[From the specs](https://w3c.github.io/csswg-drafts/mediaqueries/#pointer):

> If multiple pointing devices are present, the pointer media feature must reflect the characteristics of the “primary” pointing device

So if you plug a mouse to an iPad for example, `pointer: coarse` will still returns true since the primary input device is the touchscreen.

[any-pointer](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/any-pointer) does what you're describing, but in our case that would introduce false positive.